urllib3/util/response.py

```diff
@@ -12,18 +12,21 @@ def is_fp_closed(obj):
         The file-like object to check.
     """
 
-    try:
+    isclosed = getattr(obj, 'isclosed', None)
+    if isclosed is not None:
+        # Check isclosed() for Python 3
+        return isclosed()
+
+    closed = getattr(obj, 'closed', None)
+    if closed is not None:
         # Check via the official file-like-object way.
-        return obj.closed
-    except AttributeError:
-        pass
+        return closed
 
-    try:
+    fp = getattr(obj, 'fp', False)
+    if fp is not False:
```

Comment:
And as a future tip, it is almost always preferable, when adding function, to maintain the current style unless expressly directed not to. Style changes should ideally be done in PRs that don't change function, so that tests can accurately validate the safety of the change. =)